game-core/src/main/java/games/strategy/engine/data/DynamicallyModifiable.java

```diff
@@ -0,0 +1,21 @@
+package games.strategy.engine.data;
+
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * An interface to implement by objects that are dynamically being modified.
+ * This will most likely be related to XML-specific options.
+ */
+interface DynamicallyModifiable {
+
+  Map<String, ModifiableProperty<?>> getPropertyMap();
+
+  /**
+   * A fail-fast convenience method for {@code object.getPropertyMap().get(property)}.
+   */
+  default ModifiableProperty<?> getOrError(final String property) {
```

Comment:
* Would recommend naming this method `getPropertyOrThrow()`.
* `IllegalArgumentException` or `NoSuchElementException` might be more appropriate than `IllegalStateException`.
* Please add a `@throws` specification to the Javadocs.